PipeableTests/PipeableElement_ClickTests.swift

```diff
@@ -24,6 +24,80 @@ final class PipeableElementClickTests: PipeableXCTestCase {
         }
     }
 
+    func testBubbles() async throws {
+        let page = PipeablePage(webView)
+
+        _ = try? await page.goto("\(testServerURL)/click.html")
+
+        let clicksBefore = try await page.evaluate("window.clicksOuter")
+        guard let numClicksBefore = clicksBefore as? Int else {
+            XCTFail("Could not read clicks")
+            return
+        }
+        let focusesBefore = try await page.evaluate("window.focusesOuter")
+        guard let numFocusesBefore = focusesBefore as? Int else {
+            XCTFail("Could not read focuses")
+            return
+        }
+        let focusesInBefore = try await page.evaluate("window.focusesInOuter")
+        guard let numFocusesInBefore = focusesInBefore as? Int else {
+            XCTFail("Could not read focuses in")
+            return
+        }
+
+        let outerContainerEl = try await page.querySelector("#outer_container")
+        try await outerContainerEl?.click()
+
+        let clicksAfter = try await page.evaluate("window.clicksOuter")
+        guard let numClicksAfter = clicksAfter as? Int else {
+            XCTFail("Could not read clicks")
+            return
+        }
+        let focusesAfter = try await page.evaluate("window.focusesOuter")
+        guard let numFocusesAfter = focusesAfter as? Int else {
+            XCTFail("Could not read focuses")
+            return
+        }
+        let focusesInAfter = try await page.evaluate("window.focusesInOuter")
+        guard let numFocusesInAfter = focusesInAfter as? Int else {
+            XCTFail("Could not read focuses in")
+            return
+        }
+
+        XCTAssertEqual(numFocusesAfter, numFocusesBefore, "Focus does not buble")
```

Comment:
Not sure if I'm getting this correctly:

`focusesBefore` reads window.focusesOuter, which is triggered on `#outer_container`.
`focusesAfter` also reads window.focusesOuter, which is triggered on `#outer_container`.

We are clicking `#outer_container`. So there's no bubbling here, right? The two variables read the `focus` event on the element itself (we're not bubbling it to a parent). But then why is there no increase in the focus count?

If we want to test bubbling, we should be triggering it on an inner element and see if the parent receives it, but here we are reading and triggering `#outer_container`.

Or perhaps I'm not understanding exactly how this works?

Reply:
Good question, I added a comment and renamed the test. The logic is the following:
1) We "click" on the `outer_container` using Pipeable, but the way the click logic is implemented we find the innermost element (the link) and emit the click from there.
2) Because we are emitting the click from the innermost element (the link), there is a question about where we should emit the focus event from, should it be on the innermost element that we launched the click event from or should it be on the `outer_container` that we referenced when we did the initial click. Right now we emit the focus event from the innermost element along with the click, and I just double checked and this is identical to the behavior of playwright (atleast for this specific test).
3) Since we are emitting the click and the focus events from the innermost element (even though we are clicking on `outer_container`), and focus events do not bubble (but focusin and clicks do) we expect a focus event on the link and no `focus` event on the `outer_container`.

So really this test was very poorly named, my bad, I renamed it to `testClickAndFocusLaunchFromSameElement` since that it really what we are testing here.

Reply:
Ah, of course! 

Thanks for adding the comment!